modules/infra/control/vlan.c

```diff
@@ -154,8 +164,10 @@ static int iface_vlan_init(struct iface *iface, const void *api_info) {
 		iface, IFACE_SET_ALL, iface->flags, iface->mtu, iface->vrf_id, api_info
 	);
 	if (ret < 0) {
+		ret = errno;
 		iface_vlan_fini(iface);
-		errno = -ret;
+		errno = ret;
+		ret = -ret;
```

Comment:
I'm not sure what's happening here ?

Reply:
The idea was to preserve errno. The iface_reconfig() function may return -1 which would be mistaken as EPERM. I'll rework this to make it prettier.